jujugui/static/gui/src/app/app.js

```diff
@@ -845,20 +845,24 @@ YUI.add('juju-gui', function(Y) {
       credentials, useMacaroons, apis=[this.env, this.controllerAPI]) {
       if (useMacaroons) {
         apis.forEach(api => {
-          api.loginWithMacaroon(new Y.juju.environments.web.Bakery({
-            webhandler: new Y.juju.environments.web.WebHandler(),
-            interactive: true,
-            serviceName: 'juju',
-            dischargeStore: window.localStorage
-          }), this._apiLoginHandler.bind(this, api));
+          if (api) {
```

Comment:
It's possible that this method is called without specifying the apis. At which point the default values are used. It is however possible at that point that there is no controllerAPI instance because we are in a Juju model which does not support it. This causes the apis array to be `[this.env, undefined]` causing the login's to fail.